### PR TITLE
fix(dogstatsd): Do not egress non-finite values to Datadog Intake API

### PR DESCRIPTION
**Pull request by blt**
## Summary

This commit resolves a differential equivalence with Datadog Agent by
choosing not to egress non-finite values to the intake API. The means
here mimic Datadog Agent: allow non-finite values to be aggregated and
block them only at the end.

I did spend a chunk of time looking to not aggregate non-finite values but
in practice catching all the edge cases is branch expensive and this
solution struck me as more and more tidy as I went along. I've introduced
two tests to demonstrate the approach, one for the v2 and one for the v1
egress path.

## Change Type
- [x] Bug fix
- [ ] New feature
- [ ] Non-functional (chore, refactoring, docs)
- [ ] Performance

## How did you test this PR?

Ran an antithesis shot, confirmed no extant examples of the non-finite violation.

## References

REF SMPTNG-753

### CODE REVIEW

**Comment by blt** [comment]
* **#1924** <a href="https://app.graphite.com/github/pr/DataDog/saluki/1924?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1923** <a href="https://app.graphite.com/github/pr/DataDog/saluki/1923?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/DataDog/saluki/1923?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: lib/saluki-components/src/encoders/datadog/metrics/mod.rs [hunk 1]
```diff
@@ -967,6 +967,11 @@ fn encode_series_v2_metric(
         let value = maybe_interval
             .map(|interval| value / interval.as_secs_f64())
             .unwrap_or(value);
+
+        if !point_value_emittable(metric, value) {
+            continue;
```
[the remark this answers is not in the document]
Comment (blt) [reply lib/saluki-components/src/encoders/datadog/metrics/mod.rs]: Ah actually this is a good catch.

**Review by tobz — APPROVED** [review]
LGTM
